Update Kit and Tracks pods to versions using UIDeviceIdentifier 2.0.0

mokagio opened the pull request:
PR to verify that the app builds and the tests pass with bleeding edge versions of the Kit and Tracks pods, which have both been update to use UIDeviceIdentifier 2.0.0.

- Kit PR: https://github.com/wordpress-mobile/WordPressKit-iOS/pull/479
- Tracks PR:  https://github.com/Automattic/Automattic-Tracks-iOS/pull/193

### Testing

I’m a bit unsure about the Tracks update: [between 0.9.1 (currently on WP iOS) and 0.10.1 (latest)](https://github.com/Automattic/Automattic-Tracks-iOS/compare/0.9.1...0.10.1) we added SPM support and changed the project structure. What’s the best way to test that everything works as expected in WP after upgrading it?

By building the app and running the tests this PR provides a minimum integration test, but I think we should do additional checks before moving forward.

## Regression Notes
1. Potential unintended areas of impact:
Tracks, see above.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
T.B.D

3. What automated tests I added (or what prevented me from doing so)
None, yet.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

mokagio committed:
Point Kit and Tracks pods to WIP version using UIDeviceIdentifier 2.0.0

Notice the `TracksSetLoggingLevel` call deletion. That method is no
longer available and the whole CocoaLumberjack dependency, which that
code hit, has been removed in 0.10.0.

See
https://github.com/Automattic/Automattic-Tracks-iOS/commit/aaf79e26967635a1732fa2f2d480e060e000420f

mokagio commented:
The actual app build failed on both Buildkite and Circle CI, but the logs don't provide much info about the cause.

I'm stuck with a local Swift Package issue, the one where Xcode can't find a local package and there's no deep clean or SPM cache rebuild that helps. I've decided to take the occasion to upgrade to 13.2. I'll come to this later today.

ScoutHarris commented:
Hey @mokagio . I merged the WPKit [PR](https://github.com/wordpress-mobile/WordPressKit-iOS/pull/476) for the `feature/17135-notifications_mark_all_read_support` branch, so this should be able to use WPKit `4.47.0-beta` now.

mokagio commented on WordPress/Classes/System/WordPressAppDelegate.swift:
```diff
@@ -700,7 +700,6 @@ extension WordPressAppDelegate {
 
     @objc class func setLogLevel(_ level: DDLogLevel) {
         WPSharedSetLoggingLevel(level)
-        TracksSetLoggingLevel(level)
```
In an internal Slack, @bjhomer pointed me towards the only likely breaking change between 0.9.x and 0.10.x, that is, that CocoaLumberjack has been removed.

The Tracks README also [explains](https://github.com/Automattic/Automattic-Tracks-iOS/tree/aaa60ea7e52e1bf6eb15f1511763d694566e91dc#logging) what to do to forwards logs coming from the library to the app: create an object conforming to [`TracksLoggingDelegate`](https://github.com/Automattic/Automattic-Tracks-iOS/blob/aaa60ea7e52e1bf6eb15f1511763d694566e91dc/Sources/Model/ObjC/Common/TracksLogging.h#L5-L13) and set it as `TracksLogging.delegate`. 

That should be rather straightforward, as the delegate has only a handful of methods, all of which can share the same implementation. I have two questions about how to implement it, though:

- What's the standard logging approach in WordPress iOS?
- Should we implement one at all, or go on without it? Has anyone looked at the logs searching for Tracks related info recently?

mokagio replied:
> What's the standard logging approach in WordPress iOS?

I had a quick look around and I'm guessing CocoaLumberjack might be it. I decided to use it for the first pass at the delegate. See 0a011cc4073fad423cecb5897e35f1e2541cd327 and `TracksLogger` below.

mokagio committed:
Add new delegate for `TracksLogging`

Required as of 0.10.0 to customize logging. For the moment, it simply
calls the CocoaLumberjack log method matching the given level.

See
https://github.com/wordpress-mobile/WordPress-iOS/pull/17810/files#r792101456

mokagio commented on WordPress/WordPress.xcodeproj/project.pbxproj:
```diff
@@ -521,6 +521,8 @@
 		3F2F856326FAF612000FCDA5 /* EditorGutenbergTests.swift in Sources */ = {isa = PBXBuildFile; fileRef = CC2BB0CF228ACF710034F9AB /* EditorGutenbergTests.swift */; };
 		3F3087C424EDB7040087B548 /* AnnouncementCell.swift in Sources */ = {isa = PBXBuildFile; fileRef = 3F3087C324EDB7040087B548 /* AnnouncementCell.swift */; };
 		3F30E50923FB362700225013 /* WPTabBarController+MeNavigation.swift in Sources */ = {isa = PBXBuildFile; fileRef = 3F30E50823FB362700225013 /* WPTabBarController+MeNavigation.swift */; };
+		3F39C93527A09927001EC300 /* TracksLogger.swift in Sources */ = {isa = PBXBuildFile; fileRef = 3F39C93427A09927001EC300 /* TracksLogger.swift */; };
+		3F39C93627A09927001EC300 /* TracksLogger.swift in Sources */ = {isa = PBXBuildFile; fileRef = 3F39C93427A09927001EC300 /* TracksLogger.swift */; };
```
This is double because of the two WordPress and Jetpack targets. As someone not coding and adding files to the app on a regular basis, this never fails to surprise me 🙃

mokagio committed:
Point tracks from WIP branch to latest 0.11.0 release

ScoutHarris reviewed (approved):
Hey @mokagio .

I was able to build and run the app successfully. 🎉 I also navigated around the app and I do see tracks being logged. I'm not sure what else to test there, but it seems to be working fine. LGTM!

:shipit:

mokagio commented:
Thank you very much for taking the time @ScoutHarris 🙇‍♂️ 💙
